diff --git a/shurly.module b/shurly.module
index eec6ae9..681e2c3 100644
--- a/shurly.module
+++ b/shurly.module
@@ -236,7 +236,7 @@ function shurly_confirm_delete_form($form, &$form_state, $rid) {
       '#type' => 'value',
       '#value' => $rid,
     );
-    return confirm_form($form, t('Are you sure you want to delete and deactivate this URL?'), rawurldecode($_GET['destination']), t('You are about to deactivate the link which redirects to %url. Once this item is deleted, you will not be able to create another link with the same short URL.', array('%url' => $destination)));
+    return confirm_form($form, t('Are you sure you want to delete and deactivate this URL?'), rawurldecode($_GET['destination']), t('You are about to deactivate the link which redirects to %url. Once this item is deleted, the current date and time are appended to the Short URL for record keeping.', array('%url' => $destination)));
 }
 
 /**
@@ -244,6 +244,7 @@ function shurly_confirm_delete_form($form, &$form_state, $rid) {
  */
 function shurly_confirm_delete_form_submit($form, &$form_state) {
   drupal_set_message(t('URL has been deactivated'));
+  shurly_update_url($form_state['values']['rid']); /* This should update the URL before permanently deactivating it */
   shurly_set_link_active($form_state['values']['rid'], 0);
 }
 
@@ -656,6 +657,23 @@ function shurly_save_url($long_url, $short_path, $account = NULL, $custom = NULL
 
 
 /**
+ * API function to update a URL
+ * 
+ */
+function shurly_update_url($rid) {
+  $record = db_query('SELECT * FROM {shurly} WHERE rid = :rid', array('rid' => $rid))->fetchObject();
+  if ($record) {
+    $record->rid = $rid;
+    $record->source = $record->source . date('-Y-m-d-G-i-s'); /* appends a date to the end of the short_url */
+    return drupal_write_record('shurly', $record, 'rid');
+  }
+  else {
+    return FALSE;
+  }
+}
+
+
+/**
  * Activate or deactivate a link
  */
 function shurly_set_link_active($rid, $active) {
